src/api/app/models/project.rb

```diff
@@ -170,6 +170,18 @@ def subprojects
     Project.where("name like ?", "#{name}:%")
   end
 
+  def siblings
+    p = parent
+    siblings = []
+    unless p.nil?
+      Project.where("name like ?", "#{p.name}:%").each do |sib|
+        sib_parent = sib.possible_ancestor_names.first
+        siblings << sib if sib.name != name && sib_parent == p.name
```

Comment:
the != here sounds like a job for the database - i.e. put it as part of the where. Then your loop only has to map out those that have more :s than the project.